### ISSUE TITLE
equivalent of torch.no_grad

### ISSUE THREAD

**Issue by linyuhongg** [google/jax#9997]
I know there a function to stop the gradient from flowing down, i.e., `jax.lax.stop_gradient`. However, this function is not as flexible as `torch.no_grad`. Consider the following use case in PyTorch:

```python
def loss(x):
    features = backbone(x)
    update_actor(features.detach) # no gradient flowing down from actor network to backbone
    update_critic(features)

def update_critic(features):
    with torch.no_grad():
        action = actor(features) # no gradient flowing down from critic to actor network
    Q = critic_loss(features, action)
    loss = TD_loss(Q, target)
    loss.backward()
    optimizer.step()
```

I cannot think of anyways to perform this in JAX without having to separate the computation and perform two forward passes on the backbone. That is, if I use `jax.lax.stop_gradient` in `update_critic` it would stop the gradient from flowing down to the backbone too, since it has been stopped in the actor
